dwave/system/testing.py

```diff
@@ -64,32 +135,78 @@ class MockDWaveSampler(dwave.system.samplers.DWaveSampler):
     
     nodelist = None
     edgelist = None
     properties = None
     parameters = None
-    def __init__(self, broken_nodes=None, broken_edges=None,
-                 topology_type='chimera',topology_shape=None,
+    def __init__(self,
+                 nodelist=None, edgelist=None, properties=None,
+                 broken_nodes=None, broken_edges=None,
+                 topology_type=None,topology_shape=None,
                  parameter_warnings=True, **config):
         self.parameter_warnings = parameter_warnings
-        
-        shape_defaults = {'chimera' : [4,4,4],
-                          'pegasus' : [3],
-                          'zephyr' : [2,4]}
-        if topology_type in shape_defaults:
-            if topology_shape is None:
-                topology_shape = shape_defaults[topology_type]
+
+        #Parse or default topology dependent arguments:
+        if properties is not None and 'topology' in properties:
+            if ('type' not in properties['topology']
+                or 'shape' not in properties['topology']):
+                raise ValueError("'shape' and 'type' should be keys in "
+                                 "properties['topology']")
+            topology_type = properties['topology']['type']
+            topology_shape = properties['topology']['shape']
         else:
-            raise ValueError("Only 'chimera', 'pegasus' and 'zephyr' "
-                             "topologies are supported")
+            if topology_type is None:
+                topology_type = 'chimera'
+            shape_defaults = {'chimera' : [4,4,4],
+                              'pegasus' : [3],
+                              'zephyr' : [2,4]}
+            if topology_type in shape_defaults:
+                if topology_shape is None:
+                    topology_shape = shape_defaults[topology_type]
+            else:
+                raise ValueError("Only 'chimera', 'pegasus' and 'zephyr' "
+                                 "topologies are supported")
         self.properties = {
             'chip_id' : 'MockDWaveSampler',
-            'topology' : {'type': topology_type, 'shape': topology_shape}
-        }
-        solver_graph = self.to_networkx_graph()
+            'topology' : {'type': topology_type, 'shape': topology_shape}}
+        
+        #Create graph object, introduce defects per input arguments
+        if nodelist is not None:
+            self.nodelist = nodelist.copy()
+        if edgelist is not None:
+            self.edgelist = edgelist.copy()
+        # Note that self.to_networkx_graph would point to an inherited
+        # version rather than the class method here, without topology
+        # information, for clarity helper function is separated.
+        solver_graph = _dnx_graph(self.properties['topology']['type'],
+                                  self.properties['topology']['shape'],
+                                  self.nodelist, self.edgelist)
```

Comment:
Why not just:
```suggestion
        solver_graph = DWaveSampler.to_networkx_graph(self)
```

Reply:
Or, if you apply the suggestion for local `to_networkx_graph` below, simply:
```suggestion
        solver_graph = self.to_networkx_graph()
```

Reply:
Doesn't work as to_networkx_graph lacks topology information when inherited from dimod.Structured